code/modules/power/reactor/rbmk.dm

```diff
@@ -0,0 +1,1011 @@
+	set waitfor = FALSE
+	SSair.atmos_machinery -= src //Annd we're now just a useless brick.
+	slagged = TRUE
+	color = null
+	update_icon()
+	STOP_PROCESSING(SSmachines, src)
+	icon_state = "reactor_slagged"
+	AddComponent(/datum/component/radioactive, 15000 , src, 0)
+	var/obj/effect/landmark/nuclear_waste_spawner/NSW = new /obj/effect/landmark/nuclear_waste_spawner/strong(get_turf(src))
+	relay('sound/effects/rbmk/meltdown.ogg', "<span class='userdanger'>You hear a horrible metallic hissing.</span>")
+	stop_relay(CHANNEL_REACTOR_ALERT)
+	NSW.fire() //This will take out engineering for a decent amount of time as they have to clean up the sludge.
+	for(var/obj/machinery/power/apc/apc in GLOB.apcs_list)
+		if((apc.z == z) && prob(70))
+			apc.overload_lighting()
+	var/datum/gas_mixture/coolant_input = COOLANT_INPUT_GATE
+	var/datum/gas_mixture/moderator_input = MODERATOR_INPUT_GATE
+	var/datum/gas_mixture/coolant_output = COOLANT_OUTPUT_GATE
+	var/turf/T = get_turf(src)
+	coolant_input.set_temperature(CELSIUS_TO_KELVIN(temperature)*2)
+	moderator_input.set_temperature(CELSIUS_TO_KELVIN(temperature)*2)
+	coolant_output.set_temperature(CELSIUS_TO_KELVIN(temperature)*2)
+	T.assume_air(coolant_input)
+	T.assume_air(moderator_input)
+	T.assume_air(coolant_output)
+	explosion(get_turf(src), 0, 5, 10, 20, TRUE, TRUE)
+	empulse(get_turf(src), 25, 15)
+	QDEL_NULL(soundloop)
+
+//Failure condition 2: Blowout. Achieved by reactor going over-pressured. This is a round-ender because it requires more fuckery to achieve.
+/obj/machinery/atmospherics/components/trinary/nuclear_reactor/proc/blowout()
+	explosion(get_turf(src), GLOB.MAX_EX_DEVESTATION_RANGE, GLOB.MAX_EX_HEAVY_RANGE, GLOB.MAX_EX_LIGHT_RANGE, GLOB.MAX_EX_FLASH_RANGE)
+	meltdown() //Double kill.
+	SSweather.run_weather("nuclear fallout")
+	for(var/X in GLOB.landmarks_list)
+		if(istype(X, /obj/effect/landmark/nuclear_waste_spawner))
+			var/obj/effect/landmark/nuclear_waste_spawner/WS = X
+			if(is_station_level(WS.z)) //Begin the SLUDGING
+				WS.fire()
+
+/obj/machinery/atmospherics/components/trinary/nuclear_reactor/update_icon()
+	icon_state = "reactor_off"
+	switch(temperature)
+		if(0 to 200)
+			icon_state = "reactor_on"
+		if(200 to RBMK_TEMPERATURE_OPERATING)
+			icon_state = "reactor_hot"
+		if(RBMK_TEMPERATURE_OPERATING to 750)
+			icon_state = "reactor_veryhot"
+		if(750 to RBMK_TEMPERATURE_CRITICAL) //Point of no return.
+			icon_state = "reactor_overheat"
+		if(RBMK_TEMPERATURE_CRITICAL to INFINITY)
+			icon_state = "reactor_meltdown"
+	if(!has_fuel())
+		icon_state = "reactor_off"
+	if(slagged)
+		icon_state = "reactor_slagged"
+
+
+//Startup, shutdown
+
+/obj/machinery/atmospherics/components/trinary/nuclear_reactor/proc/start_up()
+	START_PROCESSING(SSmachines, src)
+	desired_k = 1
+	set_light(10)
+	var/startup_sound = pick('sound/effects/ship/reactor/startup.ogg', 'sound/effects/ship/reactor/startup2.ogg')
+	playsound(loc, startup_sound, 100)
+	soundloop = new(src, TRUE)
```

Comment:
```suggestion
	soundloop = new(list(src), TRUE)
```

Reply:
Don't list it
`/datum/looping_sound/New(_parent, start_immediately=FALSE, _direct=FALSE, _skip_starting_sounds = FALSE)`